unit_tests/src/test_replica_truncate.cpp

```diff
@@ -716,47 +970,158 @@ TEST_CASE("inputs_that_will_not_work")
+        CHECK(expected_error_code == rc_replica_truncate(&comm, &truncate_doi, &output_str));
+
+        // Ensure that the returned output structure has the expected contents.
+        REQUIRE(nullptr != output_str);
+        nlohmann::json json_out;
+        REQUIRE_NOTHROW(json_out = nlohmann::json::parse(output_str));
+
+        // There should be no replica information here because the expected failure occurs in such a way that no replica
+        // information is retrieved. After all, the replica being targeted does not exist.
+        const auto error_message =
+            fmt::format("rs_replica_truncate: Error occurred resolving hierarchy or getting information for [{}]: {}",
+                        target_object.c_str(),
+                        expected_error_code);
+        const auto expected_json_output = nlohmann::json{{"message", error_message},
+                                                         {"replica_number", nlohmann::json::value_t::null},
+                                                         {"resource_hierarchy", nlohmann::json::value_t::null}};
+
+        CHECK(expected_json_output == json_out);
     }
 
     SECTION("resource_name_with_no_replica")
     {
-        addKeyVal(&truncate_doi.condInput, RESC_HIER_STR_KW, test_resc.c_str());
-        CHECK(SYS_REPLICA_DOES_NOT_EXIST == rc_replica_truncate(&comm, &truncate_doi, &output_str));
+        constexpr auto expected_error_code = SYS_REPLICA_INACCESSIBLE;
+        addKeyVal(&truncate_doi.condInput, RESC_NAME_KW, test_resc.c_str());
+        CHECK(expected_error_code == rc_replica_truncate(&comm, &truncate_doi, &output_str));
+
+        // Ensure that the returned output structure has the expected contents.
+        REQUIRE(nullptr != output_str);
+        nlohmann::json json_out;
+        REQUIRE_NOTHROW(json_out = nlohmann::json::parse(output_str));
+
+        // There should be no replica information here because the expected failure indicates that the replica we wish
+        // to truncate does not exist. Even though an existing replica may have won the vote, it was not the intended
+        // target of the truncation and has nothing to do with the failure. Therefore, we expect no replica information.
+        const auto error_message = fmt::format("rs_replica_truncate: Hierarchy descending from specified resource name "
+                                               "[{}] does not have a replica of [{}] "
+                                               "or the replica is inaccessible at this time.",
+                                               test_resc.c_str(),
+                                               target_object.c_str());
+        const auto expected_json_output = nlohmann::json{{"message", error_message},
+                                                         {"replica_number", nlohmann::json::value_t::null},
+                                                         {"resource_hierarchy", nlohmann::json::value_t::null}};
```

Comment:
This is the assertion that I wanted to draw attention to.

Reply:
The new behavior is correct to me.

User requests a replica which doesn't exist or isn't under the target resource, therefore they get an error and no replica information.

Reply:
Great, that's good to hear. Thanks!

Reply:
Resolving as no objections have been raised to this as the expected behavior.